compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -301,6 +301,36 @@ public STNode parseAsObjectMember() {
         return objectMember;
     }
 
+    /**
+     * Completely parses a given input as an intermediate clause.
+     *
+     * @return Parsed node
+     */
+    public STNode parseAsIntermediateClause(boolean isRhsExpr, boolean allowActions) {
+        startContext(ParserRuleContext.COMP_UNIT);
+        startContext(ParserRuleContext.FUNC_DEF);
+        startContext(ParserRuleContext.FUNC_BODY_BLOCK);
+        startContext(ParserRuleContext.VAR_DECL_STMT);
+        startContext(ParserRuleContext.QUERY_EXPRESSION);
```

Comment:
We should give the consumer to make `allowActions` true or false. Depending on the value for `allowActions` we may need to start different set of context for module level @lochana-chathura

Reply:
contexts don't matter for the actions neh. Error handler support both. Only at the parser level do we do the invalidation?

Reply:
We have an issue differentiating this in the error handler https://github.com/ballerina-platform/ballerina-lang/issues/36387

Reply:
Current contexts can applicable for with and without allow actions. 
Do we need to start different set of context for module level?
@rdulmina

Reply:
As discussed offline this is okay